### PR TITLE
test: add tests for usage component

### LINKED ISSUE #1868: Add Tests for Usage, Overview, Readme Components.
All components currently have tests except `Usage`, `Overview`, and `Readme` packages/components.

Even though these components are primarily used to generate the README, they are still part of the public surface and are gonna export via index.js.

For consistency and to avoid regressions, it would make sense to add basic tests for these components as well. This would ensure they render the expected output and continue to behave correctly as the codebase evolves.

### PR DESCRIPTION
**Pull request by batchu5**
Summary

Improve validation and test coverage for the Usage component.

What changed
- Added a new test file to verify Usage component behavior for edge cases such as null, undefined, empty values, and incorrect casing of the language parameter.
- Updated the Usage component to handle these edge cases consistently.

**Related issue(s)**
Fixes #1868

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test suite for the Usage component to validate rendering with supported languages and error handling for invalid parameters.

### CODE REVIEW

**Comment by batchu5** [comment]
@Adi-204 I think CI failures are occurring in `@asyncapi/generator-react-sdk` tests

**Comment by SHUBHANSHU602** [comment]
@batchu5 , go through once  the failing checks shown by codebbit ai

**Comment by batchu5** [comment]
Thanks for the suggestion @SHUBHANSHU602, but I have added the docstrings in [this PR](https://github.com/asyncapi/generator/pull/1767). So I think there is no need to add them here, if it gets merged this checks will pass.

File: packages/components/src/components/readme/Usage.js [hunk 1]
```diff
@@ -31,7 +31,18 @@ main();
...
+  }
+
   const snippetFn = usageConfig[language];
+
+  if (!snippetFn) {
+    throw new Error(
+      `Invalid "language" parameter: unsupported value "${language}"`
+    );
+  }
+
```
Comment (Adi-204) [inline packages/components/src/components/readme/Usage.js]: no need to add error handling for now we already have a separate issue, revert this changes.

File: packages/components/test/components/Usage.test.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        <Usage
+          clientName="MyClient"
+          clientFileName="my_client.py"
+        />
+      )
+    ).toThrow(
+      'Invalid "language" parameter: must be a non-empty string, received undefined'
+    );
+  });
+});
```
Comment (Adi-204) [inline packages/components/test/components/Usage.test.js:30]: Also remove testcases of error handling.
